src/test/unit/middlewares/error.spec.js

```diff
@@ -0,0 +1,141 @@
+const errorMiddleware = require("~/middlewares/error")
+const {
+    INTERNAL_SERVER_ERROR,
+    DOCUMENT_ALREADY_EXISTS,
+    MONGO_SERVER_ERROR,
+    VALIDATION_ERROR
+} = require('~/consts/errors')
+const {
+    expectErrorStatus,
+    expectJSON,
+    expectLogger
+} = require("~/test/helpers")
+const logger = require('~/logger/logger')
+const getUniqueFields = require('~/utils/getUniqueFields')
+
+jest.mock('~/logger/logger')
+jest.mock('~/utils/getUniqueFields')
+
+describe("Error middleware", () => {
+    let res, next;
+
+    const expectErrorHandling = (
+        res,
+        logger,
+        err,
+        expectedStatus,
+        expectedCode,
+        expectedMessage
+    ) => {
+        expectErrorStatus(res.status, expectedStatus);
+        expectJSON(res.json, {
+            status: expectedStatus,
+            code: expectedCode,
+            message: expectedMessage
+        });
+        expectLogger(logger.error, err);
+    };
+
+
+    beforeEach(() => {
+        res = {
+            status: jest.fn().mockReturnThis(),
```

Comment:
why do we need mockReturnThis() here?

Reply:
To support chaining, for instance: `res.status(400).json({ error: 'bad_request' })`. If we didn't return **this** for both _status_ and _json_, command above would fail in tests

Reply:
So essentially we can't get `res.json({ ... })` without mockReturnThis()